public/app/core/services/backend_srv.ts

```diff
@@ -142,6 +142,83 @@ export class BackendSrv implements BackendService {
     });
   }
 
+  chunkRequestId = 1;
+
+  chunked(options: BackendSrvRequest): Observable<FetchResponse<Uint8Array | undefined>> {
+    const requestId = options.requestId ?? `chunked-${this.chunkRequestId++}`;
+    const controller = new AbortController();
+    const url = parseUrlFromOptions(options);
+    const init = parseInitFromOptions({
+      ...options,
+      requestId,
+      abortSignal: controller.signal,
+    });
+
+    return new Observable((observer) => {
+      let done = false;
+      // Calling fromFetch explicitly avoids the request queue
+      const sub = this.dependencies.fromFetch(url, init).subscribe({
+        next: (response) => {
+          const rsp = {
+            status: response.status,
+            statusText: response.statusText,
+            ok: response.ok,
+            headers: response.headers,
+            url: response.url,
+            type: response.type,
+            redirected: response.redirected,
+            config: options,
+            traceId: response.headers.get(GRAFANA_TRACEID_HEADER) ?? undefined,
+            data: undefined,
+          };
+
+          if (!response.body) {
+            observer.next(rsp);
+            observer.complete();
+            return;
+          }
+
+          const reader = response.body.getReader();
+          async function process() {
+            while (reader && !done) {
+              if (controller.signal.aborted) {
+                reader.cancel(controller.signal.reason);
+                console.log(requestId, 'signal.aborted');
+                return;
+              }
+              const chunk = await reader.read();
+              observer.next({
+                ...rsp,
+                data: chunk.value,
+              });
+              if (chunk.done) {
+                done = true;
+                console.log(requestId, 'done');
+              }
+            }
+          }
+          process()
+            .catch((e) => {
+              console.log(requestId, 'catch', e);
+            }) // from abort
+            .then(() => {
+              console.log(requestId, 'complete');
+              observer.complete();
+            }); // runs in background
+        },
```

Comment:
```suggestion
          process()
            .then(() => {
              console.log(requestId, 'complete');
              observer.complete();
            }) // runs in background
            .catch((e) => {
              console.log(requestId, 'catch', e);
              observer.error(e);
            }); // from abort
        },
```

I think we need to switch the order of `then/catch`, otherwise the errors in `then` won't be caught. Also a good idea to forward the error to observer so it's notified when `reader.read()` fails.